docs/dev.md

```diff
@@ -93,7 +93,18 @@ Your project must have a Dockerfile to use dev mode in container mode. A sample
 
 Note: Open Liberty images with the `kernel-slim` tag are not supported with dev mode at this time.
 
-Dev mode works with a temporary modified copy of your Dockerfile to allow for hot deployment during development. When dev mode starts up, it pulls the latest version of the parent image defined in the Dockerfile, builds the container image, then runs the container. Note that the context of the `docker build` command used to generate the container image is the directory containing the Dockerfile. When dev mode exits, the container is stopped and deleted, and the logs are preserved in the directory mentioned above.
+Dev mode works with a temporary, modified copy of your Dockerfile to allow for hot deployment during development as detailed below. When dev mode starts up, it pulls the latest version of the parent image defined in the Dockerfile, builds the container image, then runs the container. Note that the context of the `docker build` command used to generate the container image is the directory containing the Dockerfile. When dev mode exits, the container is stopped and deleted, and the logs are preserved in the directory mentioned above.
+
+Dev mode modifies the Dockerfile in order to allow for hot deployment. It only supports the application under development in the current project so it relies on the `.war.xml` file generated by the application build operation. This file is functionally equivalent to the `.war` file but it references class and resource files to pick up changes. To avoid application conflicts, dev mode removes all Dockerfile commands that copy or add a `.war` file. 
+
+The `.war.xml` file is generated in the `defaultServer/apps` or the `defaultServer/dropins` directory so these directories are mounted in the container. Therefore any files that the Dockerfile may have copied into these directories in the container will not be accessible.
+
+There are other features of the Dockerfile which are not supported for hot deployment of changes. If you use any of these features to COPY or ADD files you must manually restart the container after changing the files.
+- files identified using variables in the COPY or ADD commands (e.g. `$LOCATION/file.txt`)
+- files copied from a different part of a multistage Docker build (e.g. use of the `--from` option)
+- files identified by a regular expression (e.g. `resource*.txt`)
```

Comment:
Some of this information is duplicated in the File Tracking section.  Suggest to merge this into the File Tracking section instead (if anything is missing there -- it mentions wildcards but does not mention regular expressions).

Reply:
You are correct, this should be a reference to the tracking section. Also, I think wildcard is a better description than regular expression.